src/Symfony/Component/Messenger/Bridge/Redis/Transport/RedisReceiver.php

```diff
@@ -44,7 +44,7 @@ public function get(): iterable
             return [];
         }
 
-        $redisEnvelope = json_decode($message['data'], true);
+        $redisEnvelope = json_decode($message['data']['message'] ?? '', true);
```

Comment:
What about
```suggestion
        $redisEnvelope = json_decode($message['data']['message'] ?? $message['data'] ?? '', true);
```
instead?

Reply:
It can be a solution, indeed :+1:

Reply:
How can a message end up in the queue that doesn't have this `message` key?

Reply:
~Because a message can be posted by any other system than a Symfony app, and not follow the rule to have a `message` key. The only thing that we are sure, is that a data in `$message['data']` is JSON encoded. Then, we can' be sure that it contains `body` or `headers` or something else.~

Ok, my bad, I miss something. You are right, Redis gives us mandatory a `message` key, which can contains anything (not necessarily `body` and `headers`). So your fix is good :+1:

Reply:
got it